configurator/src/main/java/org/apache/felix/configurator/impl/json/JSONUtil.java

```diff
@@ -201,7 +194,7 @@ public static ConfigurationFile readJSON(
                 int ranking = 0;
                 ConfigPolicy policy = ConfigPolicy.DEFAULT;
 
-                final Dictionary<String, Object> properties = new Hashtable<>();
+                final Dictionary<String, Object> properties = new OrderedDictionary<>();
```

Comment:
@bosschaert @rombert I think that in this case we might have to maintain full backwards compatibility.

Reply:
Yes, here we need to keep everything backward compatible. The configurator 1.0.0 has already been released.

Reply:
Thinking a little more about this, I think we should try to implement SLING-7752 without needing any changes in the Felix Configurator. The configurator is an implementation of the OSGi Configurator Spec ( https://osgi.org/specification/osgi.cmpn/7.0.0/service.configurator.html ) which doesn't say anything about ordering. While AFAIK the Felix Configurator is currently the only implementation of this spec, in theory another implementation could be made that doesn't have such ordering and this could break SLING-7752 again if it was used with the Sling Feature model.